### PR TITLE
MM-15249: cache the current Jira instance

### PR DESCRIPTION
**Pull request by levb**
The default TTL is set to 1s.

Ticket: https://mattermost.atlassian.net/browse/MM-15249

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -29,12 +30,19 @@ type externalConfig struct {
...
+const currentInstanceTTL = 1 * time.Second
+
 type config struct {
 	// externalConfig caches values from the plugin's settings in the server's config.json
 	externalConfig
 
 	// Cached actual bot user ID (derived from c.UserName)
 	botUserID string
+
+	// Cached current Jira instance. A non-0 expires indicates the presense
```
Comment (jfrerich) [inline server/plugin.go]: nit: s/presense/presence/

File: server/plugin.go [hunk 2]
```diff
@@ -29,12 +30,19 @@ type externalConfig struct {
 	Secret string `json:"secret"`
 }
 
+const currentInstanceTTL = 1 * time.Second
```
Comment (jfrerich) [inline server/plugin.go]: What does multiplying by one achieve?

Comment (levb) [reply server/plugin.go]: Just a no-op to make he code more readable.

Comment (crspeller) [inline server/plugin.go]: Why 1 second?

File: server/kv.go [hunk 3]
```diff
@@ -208,17 +214,36 @@ func (p *Plugin) DeleteJiraInstance(key string) (returnErr error) {
...
 
 	return nil
 }
 
 func (p *Plugin) LoadCurrentJIRAInstance() (Instance, error) {
+	conf := p.getConfig()
+	now := time.Now()
+
+	if now.Before(conf.currentInstanceExpires) {
+		if conf.currentInstance == nil {
```
Comment (crspeller) [inline server/kv.go]: Why would this happen? Maybe just check `conf.currentInstance == nil` above and let it try to load it if it's nil?

**Pull request merged**